### PR TITLE
kubernetes-helm-mapkubeapis: init at 0.1.0

### PR DESCRIPTION
**Pull request by jschaul**
###### Description of changes

Added [hem-mapkubeapis](https://github.com/helm/helm-mapkubeapis) helm plugin

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  vendorHash = "sha256-jqVzBRlGFhDHaiSF9AArJdt4KRCiUqUuo0CnJUTbSfE=";
+
+  # NOTE: Remove the install and upgrade hooks.
+  postPatch = ''
+    sed -i '/^hooks:/,+2 d' plugin.yaml
+  '';
+
+  checkPhase = ''
+  '';
```
Comment (flokli) [inline pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix:22]: Why is this set to an empty string? Were there some tests failing? If so, please set `doCheck` to false, and leave a comment why next to it.

File: pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    mv $out/bin $out/${pname}/
+    install -m644 -Dt $out/${pname}/config/ config/Map.yaml
+    install -m644 -Dt $out/${pname} plugin.yaml
+  '';
+
+  meta = with lib; {
+    description = "A Helm plugin to map helm release deprecated Kubernetes APIs in-place";
+    homepage = "https://github.com/helm/helm-mapkubeapis";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ flokli ];
```
Comment (flokli) [inline pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix:35]: Please add yourself as maintainer too ;-) You can do this by adding yourself to `maintainers/maintainer-list.nix` (in a previous commit), and then referring to your nick here.

**Comment by aos** [comment]
Hi there - I came across this while looking to install this helm plugin. It looks like it's gone stale, as it's been over a year now. I was wondering what the policy is for getting this in? I can open my own PR with changes and addressing the comments above.

CC: @jschaul @flokli

**Comment by jschaul** [comment]
@aos  please feel free to open your own PR, I'm currently unable to follow up on this. I'll close this one.

**Pull request closed without merging**